junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/SummaryGeneratingListener.java

```diff
@@ -123,10 +122,6 @@ public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult
 					throwable -> this.summary.addFailure(testIdentifier, throwable));
 				break;
 			}
-
-			default:
-				throw new PreconditionViolationException(
-					"Unsupported execution status:" + testExecutionResult.getStatus());
```

Comment:
Okey dokey!

error-prone originally reported a compilation error of the sort ["UnnecessaryDefaultInEnumSwitch"](http://errorprone.info/bugpattern/UnnecessaryDefaultInEnumSwitch) here, so I removed it thinking that was the end of that, but coming back to this, I realise that it's erroneous to remove it since the enclosing method is `public` and `@API(Experimental)` and it's a parameter that's being switch-ed on.

I now intend to refactor it to the following so that error-prone can perform [exhaustive enum-switch analysis](http://errorprone.info/bugpattern/MissingCasesInEnumSwitch) on the switch statement.

```java
switch (testExecutionResult.getStatus()) {
  case SUCCESSFUL: { 
    ...;
    return;
  }
  case ABORTED: { 
    ...;
    return;
  }
  case FAILED: { 
    ...;
    return;
  }
}
throw new PreconditionViolationException(...);
```